src/community/oseo/oseo-core/src/main/java/org/geoserver/opensearch/eo/store/MappingFeatureCollection.java

```diff
@@ -61,4 +70,42 @@ public void close() {
             }
         };
     }
+
+    @Override
+    public void accepts(
+            org.geotools.api.feature.FeatureVisitor visitor,
+            org.geotools.api.util.ProgressListener progress)
+            throws IOException {
+        super.accepts(visitor, progress);
```

Comment:
Nope, don't call super, or the full visit, feature by feature, will happen first, and only after you'll get the optimized one. Instead, go through the ones you can optimize and exit early, leaving the call to super as a fallback at the end, for the cases we cannot optimize. 

Also add a test to make sure the optimization actually happened, mimicking this GeoTools test:
https://github.com/geotools/geotools/blob/dad50dbda0d4204ce5e5dae099c3c1da3c096e4a/modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCAggregateFunctionOnlineTest.java#L60

Reply:
fixed